compiler/resolution/callDestructors.cpp

```diff
@@ -1004,34 +1004,52 @@ fixupDestructors() {
 }
 
 
-static void insertGlobalAutoDestroyCalls() {
-  // --ipe does not build chpl_gen_main
-  if (chpl_gen_main == NULL)
+static void ensureModuleDeinitFnAnchor(ModuleSymbol* mod, Expr*& anchor) {
+  if (anchor)
     return;
 
-  SET_LINENO(baseModule);
-
-  const char* name = "chpl__autoDestroyGlobals";
-  FnSymbol*   fn   = new FnSymbol(name);
+  SET_LINENO(mod);
+  FnSymbol* deinitFn = mod->deinitFn;
+  if (!deinitFn) {
+    deinitFn = new FnSymbol(astr("chpl__deinit_", mod->name));
+    mod->block->insertAtTail(new DefExpr(deinitFn));
+    normalize(deinitFn);
+    resolveFns(deinitFn);
```

Comment:
This is a newly-created function - in case we need it and the user did not write a deinit().
It has empty body, so resolveFns() should do very little.